.github/workflows/osx.yml

```diff
@@ -0,0 +1,49 @@
+name: OSX
+
+on:
+  push:
+    branches: '*'
+
+jobs:
+  build:
+    name: OSX
+
+    runs-on: 'macos-latest'
+    strategy:
+      fail-fast: true
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: goanpeca/setup-miniconda@v1
+      with:
+        channels: conda-forge
+        auto-update-conda: true
+
+    - name: Setup
+      shell: bash -l {0}
+      run: |
+          source ./scripts/ci/osx/setup.sh
+
+    - name: CMake
+      shell: bash -l {0}
+      run: |
+          source ../scripts/ci/osx/cmake.sh
+      working-directory: ./build
+
+    - name: Compile
+      shell: bash -l {0}
+      run: |
+          source ../scripts/ci/osx/compile.sh
+      working-directory: ./build
+
+    - name: Test
+      shell: bash -l {0}
+      run: |
+          source ../scripts/ci/osx/test.sh
+      working-directory: ./build
+
+    - name: Examples
+      shell: bash -l {0}
+      run: |
+          source ./scripts/ci/osx/examples.sh
```

Comment:
Just hoping to catch it if we break one? Why not on Windows too?

Reply:
yes, this should be extended to the windows builds. The examples were never tested on windows AFAIK.

Reply:
done.